pkg/twcc/header_extension.go

```diff
@@ -0,0 +1,50 @@
+package twcc
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/pion/interceptor"
+	"github.com/pion/rtp"
+)
+
+// ErrInvalidExtensionHeaderID indicates an invalid extension header ID
+var ErrInvalidExtensionHeaderID = errors.New("invalid extension header ID")
+
+// HeaderExtensionInterceptor adds transport wide sequence numbers as header extension to each RTP packet
+type HeaderExtensionInterceptor struct {
+	interceptor.NoOp
+	extensionHeaderID uint8
+
+	seqNrLock      sync.Mutex
+	nextSequenceNr uint16
+}
+
+// NewHeaderExtensionInterceptor returns a HeaderExtensionInterceptor which uses the given ID as the extension header ID
+func NewHeaderExtensionInterceptor(id uint8) (*HeaderExtensionInterceptor, error) {
+	if id == 0 {
+		return nil, ErrInvalidExtensionHeaderID
+	}
+	return &HeaderExtensionInterceptor{extensionHeaderID: id}, nil
```

Comment:
The `id` will need to be determined in the `Bind`

This happens when a PeerConnection is created. Since we haven't done signaling yet we will not have negotiated IDs yet.

Reply:
Makes sense. I pushed another commit where I take the `id` from the `StreamInfo` in `Bind`.